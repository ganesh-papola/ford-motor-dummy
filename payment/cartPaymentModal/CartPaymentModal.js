import React, { forwardRef, useRef } from 'react';
import ReactModal from 'react-modal';
import FakeX from '@Common/fakeX/FakeX';
import './CartPaymentModal.global.scss';
import './CartPaymentModal.scoped.scss';
import { replaceTemplateWithComponent } from '../../cart/calculator/richTextDisclaimer/RichTextDisclaimer';
import BallChainMan from '../../cart/ballAndChainMan/BallChainMan';

const FakeXRefForwarded = forwardRef((props, ref) => (
  <FakeX {...props} forwardedRef={ref} />
));
FakeXRefForwarded.displayName = 'FakeX';

const CartPaymentModal = ({
  isModalOpen = false,
  closeModal = () => {},
  zebraList,
  headerText,
  title,
  subTitle,
  bottomContent,
  ngcBottomContent,
  showOneOffPayments,
  oneOffPaymentsHeader,
  oneOffModalTitle,
  oneOffModalSubTitle,
  oneOffBottomContent,
  showBallChainMan,
  ballChainText,
  pricingCategorySummaryOneOff,
  zebraListOneOffPayments,
}) => {
  const closer = useRef(null);

  const focusCloseBtn = () => {
    closer?.current?.focus();
  };

  return (
    <>
      <ReactModal
        isOpen={isModalOpen}
        contentLabel="Pricing Summary"
        className="cart-payment-modal"
        ariaHideApp={process.env.NODE_ENV !== 'test'}
        onRequestClose={closeModal}
        onAfterOpen={() => focusCloseBtn()}
        overlayClassName="dark-modal"
      >
        <div className="inner-wrapper">
          <header className="header">
            <span>{headerText}</span>

            <FakeXRefForwarded onTrigger={closeModal} ref={closer} />
          </header>
          <div className="body">
            <BallChainMan
              className="pricing-rail"
              showBallChainMan={showBallChainMan}
              ballChainText={ballChainText}
            />
            <h2 className="title">{title}</h2>
            <h3 className="sub-title">{subTitle}</h3>
          </div>
          {zebraList}
          <div className="bottom-content">
            {ngcBottomContent
              ? replaceTemplateWithComponent({
                  string: ngcBottomContent,
                  templateDescriptor: 'disclaimer',
                }).fullContentArray
              : bottomContent}
          </div>
          {showOneOffPayments && pricingCategorySummaryOneOff.length > 0 && (
            <>
              <div className="body">
                <h2 className="title">
                  {oneOffModalTitle || oneOffPaymentsHeader}
                </h2>
                <h3 className="sub-title">{oneOffModalSubTitle}</h3>
              </div>
              {zebraListOneOffPayments}
              <div className="bottom-content">{oneOffBottomContent}</div>
            </>
          )}
        </div>
      </ReactModal>
    </>
  );
};

export default CartPaymentModal;
