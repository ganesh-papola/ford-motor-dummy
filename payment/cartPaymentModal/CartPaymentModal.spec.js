import React from 'react';
import { render } from '@testing-library/react';
import CartPaymentModal from './CartPaymentModal';
import '@testing-library/jest-dom/extend-expect';

const bottomContent = `The  is based on an Optional Final Payment and Finance APR for this pre-production vehicle. Please note the final confirmed pricing will be available in 2020 once the vehicle enters production phase.

Deposit allowance where applicable only available when financed with Ford Credit.

[†] The Price Adjustment has been applied to the Online Cash Price to ensure you have equal s for the duration of your agreement.

[‡] £10 Finance Facility Fee payable with the first instalment. £10 Purchase Fee payable with the Optional Final Payment if you wish to own the vehicle at the end of the agreement.

Further charges may be made subject to the condition of the vehicle, if the vehicle is returned at the end of the finance agreement.

Finance subject to status. Freepost Ford Credit.`;

const props = {
  headerText: 'ESTIMATED PRICING SUMMARY',
  title: 'PRICING',
  subTitle: 'Your estimated vehicle price.',
  bottomContent,
  zebraListData: [
    {
      label: 'Monthly Payment',
      value: '£699.00',
    },
    {
      label: 'Number of Monthly Payments',
      value: '36',
    },
    {
      label: 'Dealer OTR Promotional Price',
      value: '£44,000.00',
    },
  ],
  isModalOpen: true,
  showOneOffPayments: true,
  oneOffPaymentsHeader: 'One-Off Payments',
  oneOffPaymentsTooltip: 'Tooltip content for one-off payments goes here.',
  oneOffPaymentsTooltipAltText: 'Learn more about One-Off Payments',
  oneOffBottomContent: 'Bottom content for the one-off payments box.',
  pricingCategorySummaryOneOff: [
    {
      label: 'Charging',
      value: '£1,250.00',
      items: [
        { label: 'Item 1', value: '£200.00' },
        { label: 'Item 2', value: '£300.00' },
      ],
    },
    {
      label: 'Protection Plans',
      value: '£1,100.00',
      items: [{ label: 'Item 1', value: '£1,100.00' }],
    },
    { label: 'Delivery', value: '£200.00' },
  ],
};

describe('PaymentDealerIconModal', () => {
  it('should render', () => {
    const { container } = render(<CartPaymentModal {...props} />);

    expect(container).toMatchSnapshot();
  });
  it('should render empty divs when closed', () => {
    const { container } = render(
      <CartPaymentModal {...props} isModalOpen={false} />
    );
    expect(container).toMatchSnapshot();
  });
});
