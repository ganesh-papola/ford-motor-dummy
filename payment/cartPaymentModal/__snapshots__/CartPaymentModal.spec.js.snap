// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDealerIconModal should render 1`] = `<div />`;

exports[`PaymentDealerIconModal should render empty divs when closed 1`] = `<div />`;
